### PR TITLE
Implement --pixels_to_jpeg mode in djxl_ng.

### PR DESCRIPTION
**Pull request by szabadka**
This change also wires up the --jpeg_quality and --use_sjpeg flags
and fixes --num_reps > 1 mode.

Added an end-to-end test for djxl_ng.

### CODE REVIEW

File: tools/roundtrip_test.sh [hunk 1]
```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+# Copyright (c) the JPEG XL Project Authors. All rights reserved.
+#
+# Use of this source code is governed by a BSD-style
+# license that can be found in the LICENSE file.
+
+# Conformance test tooling test. This is not the JPEG XL conformance test
+# runner. This test that the tooling to generate the conformance test and the
+# conformance test runner work together.
```
Comment (mo271) [inline tools/roundtrip_test.sh:9]: Let's adapt the doc string from tooling test.

**Review by mo271 — COMMENTED** [review]
Great to have roundtrip testing. I was planning on doing this with python script instead of a shell script, in order to be able to easily compare pixels of images, but using ssimulacra for this purpose is also nice.

File: tools/roundtrip_test.sh (hunk 1 above)
Comment (szabadka) [reply tools/roundtrip_test.sh]: Done.

**Review by mo271 — APPROVED** [review]

File: tools/roundtrip_test.sh [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+trap 'retcode=$?; { set +x; } 2>/dev/null; cleanup' INT TERM EXIT
+
+roundtrip_test() {
+  local infn="${JPEGXL_TEST_DATA_PATH}/$1"
+  local encargs="$2"
+  local maxdist="$3"
+  
+  local encoder="${build_dir}/tools/cjxl"
+  local decoder="${build_dir}/tools/djxl_ng"
+  local comparator="${build_dir}/tools/ssimulacra_main"
```
Comment (mo271) [inline tools/roundtrip_test.sh:30]: In some build, like making the deb packages, `/tools/ssimulacra_main` is not build, see https://github.com/libjxl/libjxl/runs/6902801608?check_suite_focus=true

**Comment by szabadka** [comment]
#1545 fixes this. For some reason I thought the jpeg reconstruction box always comes before basic info and in the test here it was the case, but apparently not always.

**Pull request merged**